.github/workflows/firebase_deploy.yaml

```diff
@@ -17,6 +17,11 @@ jobs:
     steps:
       - uses: actions/checkout@master
 
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
       - name: Set up firebase
         run: sudo npm install -g firebase-tools --unsafe
```

Comment:
pls replace this line with `run: sudo npm install -g firebase-tools@12.9.1 --unsafe`